mmap-queue/src/main/java/org/tools4j/mmap/longQueue/api/LongPoller.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2023 tools4j.org (Marco Terzer, Anton Anufriev)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.tools4j.mmap.longQueue.api;
+
+/**
+ * Long queue poller
+ */
+public interface LongPoller extends AutoCloseable {
+    /**
+     * Result of entry polling
+     */
+    enum Result {
+        /**
+         * Entry was not available.
+         */
+        NOT_AVAILABLE,
+        /**
+         * Entry was handled and the index was advanced.
+         */
+        ADVANCED,
```

Comment:
if handler returns ADVANCE, and there is no next, what do you return?
similarly for RETREAT when first is read.

options
- return RETAINED in this case
- return BOF and EOF
- return UNAVAILABLE

Reply:
I prefer BOF and EOF, as it indicates the failure as well as the direction that was attempted